workers/data_refinery_workers/processors/salmon.py

```diff
@@ -285,33 +252,113 @@ def _run_salmon(job_context: Dict, skip_processed=SKIP_PROCESSED) -> Dict:
+        logger.info("Skipping pre-processed SalmonTools run!")
+        skip = True
 
-    try:
-        tx_out = ro.r['::']('tximport', 'tximport')(job_context['output_directory'] + 'quant.sf', type="salmon", txOut=True)
-    except RRuntimeError as e:
-        # XXX/TODO: This doesn't work yet.
-        # But, eventually, it will look something like this. ^^
+    job_context['time_start'] = timezone.now()
+    if skip:  # If this procedure should be skipped, return immediately
+        job_context['time_end'] = timezone.now()
         return job_context
 
+    command_str = "salmontools extract-unmapped -u {unmapped_file} -o {output} "
+    output_prefix = job_context["output_directory"] + "unmapped_by_salmon"
+    command_str = command_str.format(unmapped_file=unmapped_filename,
+                                         output=output_prefix)
+    if "input_file_path_2" in job_context:
+        command_str += "-1 {input_1} -2 {input_2}"
+        command_str = command_str.format(input_1=job_context["input_file_path"],
+                                         input_2=job_context["input_file_path_2"])
+    else:
+        command_str += "-r {input_1}"
+        command_str= command_str.format(input_1=job_context["input_file_path"])
+
+    logger.info("Running the following SalmonTools command: %s",
+                command_str,
+                processor_job=job_context["job_id"])
+
+    completed_command = subprocess.run(command_str.split(),
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+    job_context['time_end'] = timezone.now()
+
+    # As of SalmonTools 0.1.0, completed_command.returncode is always 0,
+    # (even if error happens).  completed_command.stderr is not totally
+    # reliable either, because it will output the following line even
+    # when the execution succeeds:
+    #  "There were <N> unmapped reads\n"
+    # in which "<N>" is the number of lines in input unmapped_names.txt.
+    #
+    # As a workaround, we are using a regular expression here to test
+    # the status of SalmonTools execution.  Any text in stderr that is
+    # not in the above format is treated as error message.
+    status_str = completed_command.stderr.decode().strip()
+    success_pattern = r'^There were \d+ unmapped reads$'
+    if re.match(success_pattern, status_str):
+        result = ComputationalResult()
+        result.command_executed = command_str
+        result.system_version = __version__
+        result.time_start = job_context['time_start']
+        result.time_end = job_context['time_end']
+        result.program_version = subprocess.run(['salmontools', '--version'],
+                                                stderr=subprocess.PIPE,
+                                                stdout=subprocess.PIPE).stderr.decode().strip()
+        result.is_ccdl = True
+        result.save()
+        job_context["result"] = result
```

Comment:
What is `job_context["result"]` for?

Reply:
I copied it from `_run_salmon` function. It seems to be a key whose value is the job result.

Reply:
Yeah I figured that might be what happened, that one doesn't seem to be used anywhere either.